### PR TITLE
feat: CLI stacking tutorial

### PR DESCRIPTION
**Pull request by yknl**
This PR adds a guide on using the CLI to participate in stacking.

Related issue: https://github.com/blockstack/stacks.js/pull/857

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
No

## Are documentation updates required?
N/A

## Testing information
Run through the guide and ensure all commands work as expected.

### CODE REVIEW

File: src/pages/understand-stacks/stacking-using-CLI.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
+---
+title: Stacking using the Stacks CLI
+description: Learn how use the Stacks CLI to participate in stacking
+experience: beginner
+duration: 10 minutes
+tags:
```
Comment (agraebe) [inline src/pages/understand-stacks/stacking-using-CLI.md:6]: this should be tutorial

File: src/pages/understand-stacks/stacking-using-CLI.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  - guide
+images:
+  sm: /images/pages/stacking-rounded.svg
+---
+
+## Introduction
+
+!> The Stacking implementation is still in development and could change in the coming weeks
+
+In this guide, you'll learn how to use the Stacks CLI to participate in stacking. The CLI is a great way to quickly try out stacking on testnet. To integrate stacking into your application, see the [stacking integration guide](/stacks-blockchain/integrate-stacking).
```
Comment (agraebe) [inline src/pages/understand-stacks/stacking-using-CLI.md:16]: - "In this tutorial"
- I believe we adopted the convention to write "Stacking" to indicate it is a product
- "see the stacking integration tutorial"

File: src/pages/understand-stacks/stacking-using-CLI.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+$ node --version
+```
+
+You will also need to install the Stacks CLI from NPM:
+
+```bash
+$ npm install @stacks/cli -g
+```
+
+## Generate A New Key Pair
```
Comment (agraebe) [inline src/pages/understand-stacks/stacking-using-CLI.md:34]: I'd apply the convnetion and indicate the step # in the heading: "Step 1: Generate an account"

- we adopted using "accounts" as opposed to wallet or key pair. I'd try to stay consistent. Introducing another term might reason confusion

File: src/pages/understand-stacks/stacking-using-CLI.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    "address": "ST1P3HXR80TKT48TKM2VTKCDBS4ET9396W0W2S3K8",
+    "btcAddress": "mqPBWmSGJhrA9x5XJC6qJtFsHnudqZ2XJU",
+    "index": 0
+  }
+}
+```
+
+We'll be using this testnet key pair to perform stacking. But first we'll need to get some testnet tokens.
+
+-> If you're stacking on mainnet, make sure you have an account with sufficient number of Stacks tokens to participate.
```
Comment (agraebe) [inline src/pages/understand-stacks/stacking-using-CLI.md:51]: maybe point out how? in one sentence, either get some tokens transferred to buy them from an exchange (linking to an exchange that allows to do so, like OkCoin)

File: src/pages/understand-stacks/stacking-using-CLI.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+## Get Testnet Tokens Using The Faucet
+
+Use the following `curl` command to request tokens from the testnet node's faucet endpoint.
+
+Note that we used the address we generated above as a parameter.
```
Comment (agraebe) [inline src/pages/understand-stacks/stacking-using-CLI.md:57]: I'd use the visual note component for that. Start the line with `-> Note that ...`

File: src/pages/understand-stacks/stacking-using-CLI.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+```bash
+curl -X POST https://stacks-node-api.xenon.blockstack.org/extended/v1/faucets/stx?address=ST1P3HXR80TKT48TKM2VTKCDBS4ET9396W0W2S3K8&stacking=true
+```
+
+## Check Balance
+
+Confirm that the faucet transaction has completed by checking the balance of your address. The `-t` flag is used to indicate testnet.
```
Comment (agraebe) [inline src/pages/understand-stacks/stacking-using-CLI.md:65]: Would like to the CLI guide or references here for more details

File: src/pages/understand-stacks/stacking-using-CLI.md [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+  "unlock_height": 0,
+  "nonce": 0
+}
+```
+
+## Check Stacking Eligibility
+
+Before we send the stacking transaction, we will need to check if we're eligible for stacking.
+
+This check ensure that we meet the minimum threshold amount of STX required in order to participate in stacking.
```
Comment (agraebe) [inline src/pages/understand-stacks/stacking-using-CLI.md:81]: "ensures"

File: src/pages/understand-stacks/stacking-using-CLI.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+
+The arguments required for the `can_stack` command are:
+
+1. Amount to stack in microstacks, we'll use the entire balance in our account.
```
Comment (agraebe) [inline src/pages/understand-stacks/stacking-using-CLI.md:85]: would make this a list of bullet points as opposed to a numbered list

Comment (agraebe) [reply src/pages/understand-stacks/stacking-using-CLI.md:85]: maybe even a table right below the actual command for easier readability

File: src/pages/understand-stacks/stacking-using-CLI.md [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+3. BTC address to receive stacking rewards. We can use any valid BTC address.
+
+   `mqkccNX5h7Xy1YUku3X2fCFCC54x6HEiHk`
+
+4. The address that we will be stacking with. We'll use the address generated previously.
+
+   `ST1P3HXR80TKT48TKM2VTKCDBS4ET9396W0W2S3K8`
+
+```bash
+stx can_stack 90000000000000 10 mqkccNX5h7Xy1YUku3X2fCFCC54x6HEiHk ST1P3HXR80TKT48TKM2VTKCDBS4ET9396W0W2S3K8 -t
```
Comment (agraebe) [inline src/pages/understand-stacks/stacking-using-CLI.md:102]: would separate out the command and the response. easier to read and it allows setting the formatting for the JSON response, so it doesn't render it as bash

File: src/pages/understand-stacks/stacking-using-CLI.md [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+{ eligible: true }
+```
+
+If we meet the conditions to participate in stacking, the command will return true.
```
Comment (agraebe) [inline src/pages/understand-stacks/stacking-using-CLI.md:106]: would highlight the response value in code: `return`

File: src/pages/understand-stacks/stacking-using-CLI.md [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+
+## Stacking
```
Comment (agraebe) [inline src/pages/understand-stacks/stacking-using-CLI.md:108]: I'd change this to the action "Stack tokens" or "Perform Stacking action" or "Participate in Stacking"

File: src/pages/understand-stacks/stacking-using-CLI.md [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+  lock_period: 10,
+  unlock_height: 3960,
+  pox_address: {
+    version: '00',
+    hashbytes: '05cf52a44bf3e6829b4f8c221cc675355bf83b7d'
+  }
+}
+```
+
+Here we can see how many microstacks are locked and when they will unlock.
```
Comment (agraebe) [inline src/pages/understand-stacks/stacking-using-CLI.md:160]: Can we finish this tutorial like we finish other tutorials:

- congratulations
- summary of learnings/achievements from this tutorial
- potential next step (ie integrate into your codebase)

File: src/pages/understand-stacks/stacking.md [hunk 13]
```diff
@@ -35,6 +35,9 @@ If you would like to implement this flow in your own wallet, exchange, or any ot
 [@page-reference | inline]
 | /understand-stacks/integrate-stacking
 
+[@page-reference | inline]
```
Comment (agraebe) [inline src/pages/understand-stacks/stacking.md:38]: could we add a similar callout in the "integrate stacking" file? the CTA could be: try it with the CLI before you integrate it

**Review by agraebe — COMMENTED** [review]
thanks 🙏🏼  left some comments to increase consistency

File: src/pages/understand-stacks/stacking-using-CLI.md (hunk 4 above)
Comment (yknl) [reply src/pages/understand-stacks/stacking-using-CLI.md:51]: I think if they are participating in stacking, they should already be knowledgeable about tokens and where they can get them.

Comment (agraebe) [reply src/pages/understand-stacks/stacking-using-CLI.md:51]: good point

**Comment by diwakergupta** [comment]
Would be great to get this merged soon, I've found myself wanting to point folks to Stacking using the CLI several times recently 🙏🏽

**Comment by agraebe** [comment]
@timstackblock confirmed it can be merged

**Review by agraebe — APPROVED** [review]

**Pull request merged**